[fields] Ensure `minutesStep` is respected by fields arrows up/down

alexfauquette opened the pull request:
Fix #9331

Will still need some tests after we agree on the behavior

flaviendelangle commented on packages/x-date-pickers/src/internals/hooks/useField/useField.ts:
```diff
@@ -17,13 +17,15 @@ import { useFieldState } from './useFieldState';
   TForwardedProps extends UseFieldForwardedProps,
-  TInternalProps extends UseFieldInternalProps<any, any, any, any>,
+  TInternalProps extends UseFieldInternalProps<any, any, any, any> &
+    Partial<TimeValidationProps<TDate>>,
```
The partial is not needed here


I'm not a fan of adding all the time validation to `useField` (which is supposed to be agnostic, how about we just go for `& { minuteStep?: number }` ?

alexfauquette replied:
Fine, then we will be able to add more properties in the future is needed

LukasTy reviewed (commented):
Nice improvement! 👌 
Two observations:
1. The <kbd>Home</kbd> or <kbd>End</kbd> shortcuts might also result in an invalid value being selected. If we are respecting `minutesStep` for <kbd>ArrowUp</kbd>, <kbd>ArrowDown</kbd>, <kbd>PageUp</kbd>, and <kbd>PageDown</kbd>, maybe it would also make sense to do the same with the mentioned shortcuts? 🤔 
2. What do you think about claping/flooring the `minutesStep` to 60 or ignoring it in case it's over 60? Anything over 60 does produce inconsistent result

alexfauquette commented:
@LukasTy I added support for <kbd>Home</kbd> and <kbd>End</kbd>.

I realised <kbd>PageUp</kbd> and <kbd>PageDown</kbd> are a bit broken if step is too high, because it increment the value 5 time faster than the arrows up/down. So if `minutsStep=15` <kbd>PageUp</kbd> does the same as <kbd>Arrow Up</kbd> because 15*5 = 75 = 60 + 15

I updated the computation such that increasing with more than 60 does not overflow the minutes values

LukasTy reviewed (approved):
Working like a charm! ❤️

EitanElbaz commented:
Much love for this change, thank you!
